api/src/main/java/org/pmiops/workbench/notebooks/LeonardoNotebooksClientImpl.java

```diff
@@ -208,12 +204,24 @@ public void updateRuntime(Runtime runtime) {
                   new LeonardoUpdateRuntimeRequest()
                       .allowStop(true)
                       .runtimeConfig(
-                          buildRuntimeConfig(
-                              runtime, userProvider.get().getClusterConfigDefault())));
+                          buildRuntimeConfig(runtime, userProvider.get().getClusterConfigDefault()))
+                      .labelsToUpsert(runtimeLabels));
           return null;
         });
   }
 
+  private Map<String, String> buildRuntimeConfigurationLabels(
+      RuntimeConfigurationType runtimeConfigurationType) {
+    Map<String, String> runtimeLabels = new HashMap<>();
+    if (runtimeConfigurationType != null) {
+      runtimeLabels.put(
+          LeonardoMapper.RUNTIME_LABEL_AOU_CONFIG,
+          LeonardoMapper.RUNTIME_CONFIGURATION_TYPE_ENUM_TO_STORAGE_MAP.get(
```

Comment:
nit: I think you can just return a singletonMap here instead of building a HashMap.

https://www.geeksforgeeks.org/collections-singletonmap-method-in-java-with-examples/